### ISSUE TITLE
Feature Request: Run example

### ISSUE THREAD

**Issue by rawler** [rust-lang/rust-enhanced#266]
Cargo supports the notion of `examples` for library code. The feature is quite poorly documented - but immensely useful when building a lib.

Any .rs-file in `examples/` is supposed to declare `fn main() {}`, and is compiled as a separate crate. (Being a perfect example of how to use the library, without the special permissions of being located in the same crate)

To run an example;

`cargo run --example <basename of .rs>`

It would be very useful to be able to run this from RustEnhanced, I.E main action "Run example..." with a list of every rs file in `examples`.

**Comment by rawler** [comment]
I currently can't find the time to provide a PR, but it might be interesting for someone else to implement.

**Comment by ehuss** [comment]
I agree that it would be nice for it to automatically pop up a list of available binaries to run.  My thought was to somehow automatically detect this scenario (either by inspecting the workspace or catching the error message when `cargo run` is used), and show a quick panel to choose the correct one.

However, you can do this today in a few ways:

### Automatic Build Variant
If you choose the "RustEnhanced - Automatic" build variant, then if you have an `example.rs` file open, running a build (Cmd/Ctrl+B) will run the example.

### Context Menu
(Right mouse button) Context Menu > Rust > Run This File.

### Custom Build Variants
You can also configure custom build variants, one for each example.  This is a more advanced method that is appropriate if you want to configure custom command-line arguments or other settings.  In short, run the Sublime command "Rust: Create New Cargo Build Variant" > Run > Choose your example > Set the name for your build variant.  Then choose the build variant (Cmd/Ctrl+Shift+B)
